src/Controls/tests/Core.UnitTests/TimePickerUnitTest.cs

```diff
@@ -39,5 +39,57 @@ public void ZeroTimeIsValid()
 				Time = new TimeSpan(0, 0, 0)
 			};
 		}
+
+		[Fact]
+		public void TestTimeSelected()
+		{
+			var picker = new TimePicker();
+
+			bool selected = false;
```

Comment:
Instead of a `bool`, could this be an `int` that is incremented in the event? This way the Assert can also verify that the event was fired only once.

Reply:
Updated, great idea!